File: Producto/VeciZervi-BackEnd/backend/src/main/java/com/vecizervi/backend/model/Resena.java
package com.vecizervi.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;

@Entity
@Table(name = "resenas")
public class Resena {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_resena")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "id_trabajo", nullable = false)
    // Solo devolver id y titulo del trabajo para evitar ciclos
    @JsonIgnoreProperties({
        "cliente","categoria","descripcion","postulaciones",
        "mensajes","estado","precio","comuna","fechaPublicacion",
        "fechaFinalizacion","latitud","longitud"
    })
    private Trabajo trabajo;

    @ManyToOne
    @JoinColumn(name = "id_emisor", nullable = false)
    @JsonIgnoreProperties({
        "password","tokenRecuperacion","cuentaBloqueadaHasta",
        "intentosFallidos","herramientasPropias","calificacionPromedio",
        "rut","fechaNacimiento","correo","rol"
    })
    private Usuario emisor;

    @ManyToOne
    @JoinColumn(name = "id_receptor", nullable = false)
    @JsonIgnoreProperties({
        "password","tokenRecuperacion","cuentaBloqueadaHasta",
        "intentosFallidos","herramientasPropias","calificacionPromedio",
        "rut","fechaNacimiento","correo","rol"
    })
    private Usuario receptor;

    @Column(nullable = false)
    private Integer estrellas;

    @Column(columnDefinition = "TEXT")
    private String comentario;

    @Column(name = "url_foto_evidencia")
    private String urlFotoEvidencia;

    public Resena() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Trabajo getTrabajo() { return trabajo; }
    public void setTrabajo(Trabajo trabajo) { this.trabajo = trabajo; }
    public Usuario getEmisor() { return emisor; }
    public void setEmisor(Usuario emisor) { this.emisor = emisor; }
    public Usuario getReceptor() { return receptor; }
    public void setReceptor(Usuario receptor) { this.receptor = receptor; }
    public Integer getEstrellas() { return estrellas; }
    public void setEstrellas(Integer estrellas) { this.estrellas = estrellas; }
    public String getComentario() { return comentario; }
    public void setComentario(String comentario) { this.comentario = comentario; }
    public String getUrlFotoEvidencia() { return urlFotoEvidencia; }
    public void setUrlFotoEvidencia(String url) { this.urlFotoEvidencia = url; }
}

File: Producto/Android Studio/VeciZervi/app/src/main/java/com/example/vecizervi/data/models/Resena.kt
package com.example.vecizervi.data.models

import com.google.gson.annotations.SerializedName

data class ResenaEmisor(
    @SerializedName("idUsuario") val idUsuario: Int = 0,
    @SerializedName("nombres")   val nombres: String = "",
    @SerializedName("apellidos") val apellidos: String = ""
)

data class Resena(
    // El backend devuelve "id" no "id_resena"
    @SerializedName("id")                 val idResena: Long? = null,
    // Campos para enviar (POST)
    @SerializedName("id_trabajo")         val idTrabajo: Long = 0,
    @SerializedName("id_emisor")          val idEmisor: Long = 0,
    @SerializedName("id_receptor")        val idReceptor: Long = 0,
    // Campos que devuelve el backend (GET)
    @SerializedName("emisor")             val emisorObj: ResenaEmisor? = null,
    @SerializedName("receptor")           val receptorObj: ResenaEmisor? = null,
    @SerializedName("estrellas")          val estrellas: Int = 5,
    @SerializedName("comentario")         val comentario: String = "",
    @SerializedName("url_foto_evidencia") val urlFotoEvidencia: String? = null
)